src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/NavigationServiceExtensions.cs

```diff
@@ -214,6 +200,24 @@ public static class NavigationServiceExtensions
             return registrationInfo;
         }
 
+
+        /// <summary>
+        /// Adds the given condition to the registration.
+        /// Every time a associated ViewModel is requested, the condition is evaluated before showing it.
+        /// </summary>
+        /// <param name="registration"></param>
+        /// <param name="condition"><see cref="Func{TResult}" />Condition to evaluate before showing</param>
+        /// <returns>Registration info</returns>
+        public static NavigationRegistration WithCondition(
+            this NavigationRegistration registration,
+            Func<bool> condition)
+        {
+            registration.Condition = condition;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'registration' before using it.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)